src/lib/ext-monitor.js

```diff
@@ -97,7 +106,31 @@ export default class ExtensionMonitor {
     }
   };
 
+  onInstalledExtension = async (extInfo) => {
+    const { enabled, id, type } = extInfo;
+
+    const self = await browser.management.getSelf();
+    const monitorStarted = this.hasActivityListeners();
```

Comment:
It shouldn't be needed to check `hasActivityListeners`. The event is not fired when the listener has been unregistered.